File: browser-use-python/src/browser_use_sdk/v2/resources/marketplace.py
from __future__ import annotations

from typing import Any, Optional

from ..._core.http import AsyncHttpClient, SyncHttpClient
from ...generated.v2.models import (
    ExecuteSkillResponse,
    MarketplaceSkillListResponse,
    MarketplaceSkillResponse,
    SkillResponse,
)


class Marketplace:
    def __init__(self, http: SyncHttpClient) -> None:
        self._http = http

    def list(
        self,
        *,
        page_size: Optional[int] = None,
        page_number: Optional[int] = None,
        category: Optional[str] = None,
        query: Optional[str] = None,
        from_date: Optional[str] = None,
        to_date: Optional[str] = None,
    ) -> MarketplaceSkillListResponse:
        return MarketplaceSkillListResponse.model_validate(
            self._http.request(
                "GET",
                "/marketplace/skills",
                params={
                    "pageSize": page_size,
                    "pageNumber": page_number,
                    "category": category,
                    "query": query,
                    "fromDate": from_date,
                    "toDate": to_date,
                },
            )
        )

    def get(self, slug: str) -> MarketplaceSkillResponse:
        return MarketplaceSkillResponse.model_validate(
            self._http.request("GET", f"/marketplace/skills/{slug}")
        )

    def clone(self, skill_id: str) -> SkillResponse:
        return SkillResponse.model_validate(
            self._http.request("POST", f"/marketplace/skills/{skill_id}/clone")
        )

    def execute(self, skill_id: str, **kwargs: Any) -> ExecuteSkillResponse:
        return ExecuteSkillResponse.model_validate(
            self._http.request(
                "POST", f"/marketplace/skills/{skill_id}/execute", json=kwargs
            )
        )


class AsyncMarketplace:
    def __init__(self, http: AsyncHttpClient) -> None:
        self._http = http

    async def list(
        self,
        *,
        page_size: Optional[int] = None,
        page_number: Optional[int] = None,
        category: Optional[str] = None,
        query: Optional[str] = None,
        from_date: Optional[str] = None,
        to_date: Optional[str] = None,
    ) -> MarketplaceSkillListResponse:
        return MarketplaceSkillListResponse.model_validate(
            await self._http.request(
                "GET",
                "/marketplace/skills",
                params={
                    "pageSize": page_size,
                    "pageNumber": page_number,
                    "category": category,
                    "query": query,
                    "fromDate": from_date,
                    "toDate": to_date,
                },
            )
        )

    async def get(self, slug: str) -> MarketplaceSkillResponse:
        return MarketplaceSkillResponse.model_validate(
            await self._http.request("GET", f"/marketplace/skills/{slug}")
        )

    async def clone(self, skill_id: str) -> SkillResponse:
        return SkillResponse.model_validate(
            await self._http.request("POST", f"/marketplace/skills/{skill_id}/clone")
        )

    async def execute(self, skill_id: str, **kwargs: Any) -> ExecuteSkillResponse:
        return ExecuteSkillResponse.model_validate(
            await self._http.request(
                "POST", f"/marketplace/skills/{skill_id}/execute", json=kwargs
            )
        )

File: browser-use-node/src/v2/resources/marketplace.ts
import type { HttpClient } from "../../core/http.js";
import type { components } from "../../generated/v2/types.js";

type ExecuteSkillRequest = components["schemas"]["ExecuteSkillRequest"];
type ExecuteSkillResponse = components["schemas"]["ExecuteSkillResponse"];
type MarketplaceSkillListResponse = components["schemas"]["MarketplaceSkillListResponse"];
type MarketplaceSkillResponse = components["schemas"]["MarketplaceSkillResponse"];
type SkillResponse = components["schemas"]["SkillResponse"];

export interface MarketplaceListParams {
  pageSize?: number;
  pageNumber?: number;
  query?: string;
  category?: string;
  fromDate?: string;
  toDate?: string;
}

export class Marketplace {
  constructor(private readonly http: HttpClient) {}

  /** List all public skills in the marketplace. */
  list(params?: MarketplaceListParams): Promise<MarketplaceSkillListResponse> {
    return this.http.get<MarketplaceSkillListResponse>(
      "/marketplace/skills",
      params as Record<string, unknown>,
    );
  }

  /** Get details of a specific marketplace skill by slug. */
  get(skillSlug: string): Promise<MarketplaceSkillResponse> {
    return this.http.get<MarketplaceSkillResponse>(`/marketplace/skills/${skillSlug}`);
  }

  /** Clone a public marketplace skill to your project. */
  clone(skillId: string): Promise<SkillResponse> {
    return this.http.post<SkillResponse>(`/marketplace/skills/${skillId}/clone`);
  }

  /** Execute a marketplace skill. */
  execute(skillId: string, body: ExecuteSkillRequest): Promise<ExecuteSkillResponse> {
    return this.http.post<ExecuteSkillResponse>(
      `/marketplace/skills/${skillId}/execute`,
      body,
    );
  }
}
